backstory/src/demo/webapp/vision-demo.js

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* exported getAnalyzedImages fetchBlobstoreUrl */
+
+
+/**
+ * Fetches the URL for uploading to Blobstore and adds it to the image upload
+ * form
+ */
+function fetchBlobstoreUrl() {
+  fetch('/blobstore-upload-url')
+      .then((response) => {
+        return response.text();
+      })
+      .then((imageUploadUrl) => {
+        const imageUploadForm = document.getElementById('image-upload');
+        imageUploadForm.action = imageUploadUrl;
+      });
+}
+
+/** Adds the  analyzed images to the image-list unordered list element */
+function getAnalyzedImages() {
+  fetch('/image-analysis')
+      .then((response) => response.json())
+      .then((analyzedImagesObject) => {
+        const imageListElement = document.getElementById('image-list');
+        imageListElement.innerHTML = '';
+
+        for (let i = 0; i < analyzedImagesObject.length; i++) {
+          const imageUrl = analyzedImagesObject[i].imageUrl;
+          const labelsJsonArray = analyzedImagesObject[i].labelsJsonArray;
+
+          imageListElement.appendChild(
+              createListElement(imageUrl, labelsJsonArray));
+        }
+      });
+}
+
+/** @return {Element} containing the analyzed image with its labels */
```

Comment:
While this does return an object of type Element, isn't the data type actually object? I was under the impression the data type went inside the {} e.g. ```@return {object} - an Element containing the analyzed image with its labels```.

Reply:
I can change this in a new PR, but it's out of the scope for this one. This PR only rearranges the file structure to put the demos into their own folder. Sorry about the confusion.

Reply:
That makes sense.